pkg/controller/account/account_controller.go

```diff
@@ -849,18 +850,24 @@ func CreateEC2Instance(client awsclient.Client) (string, error) {
 			MinCount:     aws.Int64(1),
 			MaxCount:     aws.Int64(1),
 		})
-		if runErr == nil {
-			instanceID = *runResult.Instances[0].InstanceId
-			break
+
+		// Return on unexpected errors
+		if runErr != nil {
+			aerr, ok := runErr.(awserr.Error)
```

Comment:
I'm reading through some other parts of the operator and its getting terse to read. Could you add comments into your PR, especially in this block?

Reply:
@fahlmant This block in particular already has a comment that I think pretty well encapsulates its functionality. I asked for a second, outside opinion and it was agreed. I'd be happy to go through the rest of the code and see how I could better clarify, but I think that it would be appropriate to leave that for another ticket and get this fix pushed as I have already greatly increased the level of commenting. 

I did forget to remove the print blocks used for testing, so those will go away.

Reply:
I agree that at first glance it's not obvious why we do that on line #856

I'd suggest adding a comment addressing that line, explaining we do that so aws API will add more info on top of the error, like the Code we use right below